cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsProcessor.java

```diff
@@ -76,9 +77,14 @@ private void updateCachedNumCoresByBroker(Cluster cluster) {
           LOG.warn("Received metrics from unrecognized broker {}.", bid);
           return null;
         }
-        BrokerCapacityInfo capacity = _brokerCapacityConfigResolver.capacityForBroker(getRackHandleNull(node), node.host(), bid);
-        // No mapping shall be recorded if capacity is estimated, but estimation is not allowed.
-        return (!_allowCpuCapacityEstimation && capacity.isEstimated()) ? null : capacity.numCpuCores();
+        try {
+          BrokerCapacityInfo capacity = _brokerCapacityConfigResolver.capacityForBroker(getRackHandleNull(node), node.host(), bid);
+          // No mapping shall be recorded if capacity is estimated, but estimation is not allowed.
+          return (!_allowCpuCapacityEstimation && capacity.isEstimated()) ? null : capacity.numCpuCores();
+        } catch (TimeoutException tme) {
+          LOG.warn("Unable to get core number of broker {}.", node.id());
+          return null;
```

Comment:
Can we update the JavaDoc to indicate cases that a null response is expected? Can we add a unit test to verify the return of expected `null` response?